skimage/measure/fit.py

```diff
@@ -858,8 +882,12 @@ def ransac(data, model_class, min_samples, residual_threshold,
     
     random_state = check_random_state(random_state)
 
-    if min_samples < 0:
+    if isinstance(min_samples, int) and min_samples < 0:
         raise ValueError("`min_samples` must be greater than zero")
+    elif isinstance(min_samples, float):
+        if not (0 < min_samples <= 1):
```

Comment:
Th inclusion or exclusion of zero is inconsistent with the docstring and error messages.